src/backend/marsha/core/views.py

```diff
@@ -440,23 +442,65 @@ def post(self, request, *args, **kwargs):
             generated from applying the data to the template
 
         """
-        lti = LTI(self.request)
         try:
-            lti.verify()
-        except LTIException as error:
+            jwt_token = AccessToken(self.request.POST.get("jwt", ""))
+            jwt_token.verify()
+
+            lti_select_form_data = jwt_token.payload.get("lti_select_form_data")
```

Comment:
Is it needed to set everything in the token ? It's not the purpose of a token but maybe in this case it can be use like this, can you explain me why ?

Reply:
The idea here is to prevent any malicious modification of the initial LTI request data.